python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,525 @@
+        Common interface for pipeline executor factory modules.
+    """
+
+    def __init__(self, module):
+        self.module = module.module
+
+
+class PipelineConfig(object):
+    """The wrapper of each module to be pipelined. The wrapper mainly includes the
+    module itself as well as the binding that represents the connections of this
+    module's inputs and outputs to other modules.
+    """
+
+    class Binding:
+        """This class define the module connection information.
+        The type can only be either "input" or "output".
+
+        Parameters
+        ----------
+        owner : ModuleWrapper
+            The class who owns this interface.
+
+        io_type : str
+            The type of this interface. It can only be either "input" or "output".
+
+        name : str/integer
+            Name, for input it is string such as "data0", for output it is the
+            idx integer such as 0.
+        """
+
+        def __init__(self, owner, io_type, name, data_type=None):
+            self.io_owner = owner
+            self.io_type = io_type
+            self.name = str(name)
+            # Child nodes.
+            self.bindings = []
+            # Parents nodes.
+            self.parents = []
+
+            self.data_type = data_type
+
+        def get_name(self):
+            """Return the interface name and name of owner who owns this interface."""
+            owner_name = ""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                owner_name = self.io_owner.name
+
+            return owner_name, self.name
+
+        def get_owner_idx(self):
+            """Return owner idex if owner is ModuleWrapper, if not return 0."""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                return self.io_owner.idx
+
+            # If not ModuleWrapper then owner is PipelineConfig, return 0
+            # to identify this is global interface
+            return 0
+
+        def is_global_interface(self):
+            """The global interface is the interface visible to the caller which use pipeline
+            executor, the global input interface is responsible for passing parameters to the
+            internal module interface, and the global output interface is responsible for
+            outputting the pipeline executor compute results to caller.
+            """
+            return not isinstance(self.io_owner, PipelineConfig.ModuleWrapper)
+
+        def __repr__(self):
+            """Get all binding(input data) informations that looks like '|data_0: mod1:data_0'."""
+            ret = "  |{}: ".format(self.name)
+            for binding in self.bindings:
+                mname, dname = binding.get_name()
+                ret += "{0}:{1} ".format(mname, dname)
+            return ret
+
+        def check_dag_acyclic(self, start, inputs):
+            """It is to check whether the DAG that contains the inputs interfaces is acyclic."""
```

Comment:
the inputs interfaces -> the input interface

Reply:
changed into "This is to check whether the DAG containing these input interfaces is acyclic."
here may need to keep 'interfaces' for that there are multiple input interfaces get involved by checking.